File: PHP/Endpoint Examples/Multipart Payload/unrestricted-pdf.php
<?php
require 'vendor/autoload.php'; // Require the autoload file to load Guzzle HTTP client.

use GuzzleHttp\Client; // Import the Guzzle HTTP client namespace.
use GuzzleHttp\Psr7\Request; // Import the PSR-7 Request class.
use GuzzleHttp\Psr7\Utils; // Import the PSR-7 Utils class for working with streams.

// Toggle deletion of sensitive files (default: false)
$DELETE_SENSITIVE_FILES = false;

$client = new Client(); // Create a new instance of the Guzzle HTTP client.

$headers = [
  'Api-Key' => 'xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx' // Set the API key in the headers for authentication.
];

$options = [
  'multipart' => [
    [
      'name' => 'file', // Specify the field name for the file.
      'contents' => Utils::tryFopen('/path/to/file', 'r'), // Open the file specified by '/path/to/file' for reading.
      'filename' => '/path/to/file', // Set the filename for the file to be processed, in this case, '/path/to/file'.
      'headers' => [
        'Content-Type' => '<Content-type header>' // Set the Content-Type header for the file.
      ]
    ],
    [
      'name' => 'current_permissions_password', // Specify the field name for the current_permissions_password option.
      'contents' => 'current_example_pw' // Set the value for the current_permissions_password option to 'current_example_pw'.
    ],
    [
      'name' => 'output', // Specify the field name for the output option.
      'contents' => 'pdfrest_unrestricted_pdf' // Set the value for the output option to generate an unrestricted PDF.
    ]
  ]
];

$request = new Request('POST', 'https://api.pdfrest.com/unrestricted-pdf', $headers); // Create a new HTTP POST request with the API endpoint and headers.

$res = $client->sendAsync($request, $options)->wait(); // Send the asynchronous request and wait for the response.

$body = (string) $res->getBody();
echo $body; // Output the response body

// All files uploaded or generated are automatically deleted based on the 
// File Retention Period as shown on https://pdfrest.com/pricing. 
// For immediate deletion of files, particularly when sensitive data 
// is involved, an explicit delete call can be made to the API.
//
// The following code is an optional step to delete sensitive files
// (unredacted, unencrypted, unrestricted, or unwatermarked) from pdfRest servers.

if ($DELETE_SENSITIVE_FILES) {
  $delete_client = new Client(['http_errors' => false]);
  $delete_headers = [
    'api-key' => 'xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx',
    'Content-Type' => 'application/json'
  ];
  $delete_body = '{"ids":"xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx"}';
  $delete_request = new Request('POST', 'https://api.pdfrest.com/delete', $delete_headers, $delete_body);
  $delete_res = $delete_client->sendAsync($delete_request)->wait();
  echo $delete_res->getBody() . PHP_EOL;
  $delete_client = new Client(['http_errors' => false]);
  $delete_headers = [
    'api-key' => 'xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx',
    'Content-Type' => 'application/json'
  ];
  $json = json_decode($body, true);
  $delete_body = '{"ids":"' . $json['outputId'] . '"}';
  $delete_request = new Request('POST', 'https://api.pdfrest.com/delete', $delete_headers, $delete_body);
  $delete_res = $delete_client->sendAsync($delete_request)->wait();
  echo $delete_res->getBody() . PHP_EOL;
}

File: DotNET/Endpoint Examples/JSON Payload/unrestricted-pdf.cs

/*
 * What this sample does:
 * - Called from Program.cs to remove permissions restrictions from a PDF via JSON flow.
 *
 * Setup (environment):
 * - Copy .env.example to .env
 * - Set PDFREST_API_KEY=your_api_key_here
 * - Optional: set PDFREST_URL to override the API region. For EU/GDPR compliance and proximity, use:
 *     PDFREST_URL=https://eu-api.pdfrest.com
 *   For more information visit https://pdfrest.com/pricing#how-do-eu-gdpr-api-calls-work
 *
 * Usage:
 *   dotnet run -- unrestricted-pdf /path/to/input.pdf
 *
 * Output:
 * - Prints JSON responses; non-2xx results exit non-zero.
 */
using Newtonsoft.Json.Linq;
using System.Text;

namespace Samples.EndpointExamples.JsonPayload
{
    public static class UnrestrictedPdf
    {
        public static async Task Execute(string[] args)
        {
            if (args == null || args.Length < 1)
            {
                Console.Error.WriteLine("unrestricted-pdf requires <inputFile>");
                Environment.Exit(1);
                return;
            }

            var inputPath = args[0];
            if (!File.Exists(inputPath))
            {
                Console.Error.WriteLine($"File not found: {inputPath}");
                Environment.Exit(1);
                return;
            }

            var apiKey = Environment.GetEnvironmentVariable("PDFREST_API_KEY");
            if (string.IsNullOrWhiteSpace(apiKey))
            {
                Console.Error.WriteLine("Missing required environment variable: PDFREST_API_KEY");
                Environment.Exit(1);
                return;
            }
            var baseUrl = Environment.GetEnvironmentVariable("PDFREST_URL") ?? "https://api.pdfrest.com";

            using (var httpClient = new HttpClient { BaseAddress = new Uri(baseUrl) })
            {
                using (var uploadRequest = new HttpRequestMessage(HttpMethod.Post, "upload"))
                {
                    uploadRequest.Headers.TryAddWithoutValidation("Api-Key", apiKey);
                    uploadRequest.Headers.Accept.Add(new("application/json"));

                    var uploadByteArray = File.ReadAllBytes(inputPath);
                    var uploadByteAryContent = new ByteArrayContent(uploadByteArray);
                    uploadByteAryContent.Headers.TryAddWithoutValidation("Content-Type", "application/octet-stream");
                    uploadByteAryContent.Headers.TryAddWithoutValidation("Content-Filename", Path.GetFileName(inputPath));


                    uploadRequest.Content = uploadByteAryContent;
                    var uploadResponse = await httpClient.SendAsync(uploadRequest);

                    var uploadResult = await uploadResponse.Content.ReadAsStringAsync();

                    Console.WriteLine("Upload response received.");
                    Console.WriteLine(uploadResult);

                    JObject uploadResultJson = JObject.Parse(uploadResult);
                    var uploadedID = uploadResultJson["files"][0]["id"];
                    using (var unrestrictRequest = new HttpRequestMessage(HttpMethod.Post, "unrestricted-pdf"))
                    {
                        unrestrictRequest.Headers.TryAddWithoutValidation("Api-Key", apiKey);
                        unrestrictRequest.Headers.Accept.Add(new("application/json"));

                        unrestrictRequest.Headers.TryAddWithoutValidation("Content-Type", "application/json");


                        JObject parameterJson = new JObject
                        {
                            ["id"] = uploadedID,
                            ["current_permissions_password"] = "password",
                        };

                        unrestrictRequest.Content = new StringContent(parameterJson.ToString(), Encoding.UTF8, "application/json"); ;
                        var unrestrictResponse = await httpClient.SendAsync(unrestrictRequest);

                        var unrestrictResult = await unrestrictResponse.Content.ReadAsStringAsync();

                        Console.WriteLine("Processing response received.");
                        Console.WriteLine(unrestrictResult);

                        // All files uploaded or generated are automatically deleted based on the 
                        // File Retention Period as shown on https://pdfrest.com/pricing. 
                        // For immediate deletion of files, particularly when sensitive data 
                        // is involved, an explicit delete call can be made to the API.
                        //
                        // Deletes all files in the workflow, including outputs. Save all desired files before enabling this step.

                        // Optional immediate deletion of sensitive files
                        // Default: false; override with PDFREST_DELETE_SENSITIVE_FILES=true
                        var deleteSensitiveFiles = string.Equals(
                            Environment.GetEnvironmentVariable("PDFREST_DELETE_SENSITIVE_FILES"),
                            "true",
                            StringComparison.OrdinalIgnoreCase);

                        if (deleteSensitiveFiles)
                        {
                            using (var deleteRequest = new HttpRequestMessage(HttpMethod.Post, "delete"))
                            {
                                deleteRequest.Headers.TryAddWithoutValidation("Api-Key", apiKey);
                                deleteRequest.Headers.Accept.Add(new("application/json"));
                                deleteRequest.Headers.TryAddWithoutValidation("Content-Type", "application/json");

                                var parsed = JObject.Parse(unrestrictResult);
                                var outId = parsed["outputId"].ToString();
                                JObject deleteJson = new JObject { ["ids"] = $"{uploadedID}, {outId}" };
                                deleteRequest.Content = new StringContent(deleteJson.ToString(), Encoding.UTF8, "application/json");
                                var deleteResponse = await httpClient.SendAsync(deleteRequest);
                                var deleteResult = await deleteResponse.Content.ReadAsStringAsync();
                                Console.WriteLine(deleteResult);
                            }
                        }
                    }
                }
            }
        }
    }
}